### ISSUE TITLE
Skip Transaction Tracing for ActionController::RoutingError in a Rails Exception App

### ISSUE THREAD

**Issue by silvermind** [getsentry/sentry-ruby#1830]
HI guys,

I'm trying to prevent ActionController::RoutingError **Transaction Events** from being sent to Sentry. As regular pentest scans pollute our performance reports in sentry. The default config IGNORE_DEFAULT already skips the 'ActionController::RoutingError' Error Event, but the Transaction Event is still sent with status internal_error.

Using the Sampling Function seems a possible way to prevent TX recordings, but seems not to work for 404 & other routing errors as it's loaded quite early in the stack: https://docs.sentry.io/platforms/ruby/configuration/sampling/#setting-a-sampling-function

https://docs.sentry.io/platforms/ruby/configuration/sampling/#forcing-a-sampling-decision
"Absolute decision passed to start_transaction" seems to be another way to control TX recording.

our Rails App has a configured Error Handler `config.exceptions_app = ->(env) { ExceptionsController.action(:show).call(env) }`

Is it possible to skip the Transaction Events recording from the Exception Controller somehow, while using the default rails instrumentation intact? 

Any ideas & hints would be very welcomed. Alternative would be so filter our pentesters with UserAgent or IPs, would you suggest this route instead?

**Comment by sl0thentr0py** [comment]
Hi @silvermind, we added the whole rack env to `sampling_context` so you can use it in your `traces_sampler`. See an example below. Does that help?

https://github.com/getsentry/sentry-ruby/blob/c237974b185e51296542237cbca7f9c1ba1a7eb6/sentry-rails/spec/sentry/rails/tracing_spec.rb#L209-L217

**Comment by silvermind** [comment]
worked :) thanks a lot.

would be nice to have the above sample with sampling_context[:env]  added to the docs at:
https://docs.sentry.io/platforms/ruby/configuration/sampling/

**Comment by aprescott** [comment]
I'm encountering this same issue. An application I work on has transactions from noisy requests to paths like `/wp-login.php`. I am trying to find some means of filtering them out.

Some gem info:

* rails 7.0.4.3
* sentry-ruby 5.8.0
* sentry-rails 5.8.0
* puma 5.6.5

I have tried `config.excluded_exceptions += ["ActionController::RoutingError"]`, which doesn't seem to take effect.

When booting the Rails app and making a request for a path that doesn't exist (`/does-not-exist`), the transaction for the request is submitted to Sentry with a status of `internal_error` as part of a `http.server` operation. There is seemingly insufficient metadata to identify it as a 404.

<img width="666" alt="Screenshot 2023-03-27 at 16 34 27" src="https://user-images.githubusercontent.com/342081/228060275-01dcd662-9355-49ae-b996-69277e5a0aac.png">

<img width="358" alt="Screenshot 2023-03-27 at 16 34 47" src="https://user-images.githubusercontent.com/342081/228060286-6a67536c-1191-47fb-aa69-f93180d07b64.png">

Curiously, in investigating this, I noticed that in _development_ (with `SENTRY_DSN` set) a 404'd request will submit a trace with `not_found` as a trace status, whereas in a deployed environment the 404'd path will submit a trace with `internal_error`. I'm not sure why the two environments differ in their status handling.
